src/components/TopologyPopover/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useRef, useState } from "react";
+
+import clsx from "clsx";
+import { FaCog } from "react-icons/fa";
+import { BsSortDown, BsSortUp } from "react-icons/bs";
+
+import { TopologySort } from "./topologySort";
+import { searchParamsToObj } from "../../utils/common";
+import { TopologyPreference } from "./topologyPreference";
+import { NavigateOptions, URLSearchParamsInit } from "react-router-dom";
+
+const defaultSortTypes = [
+  { id: 1, value: "status", label: "Health" },
+  { id: 2, value: "name", label: "Name" },
+  { id: 3, value: "type", label: "Type" },
+  { id: 4, value: "updated_at", label: "Last Updated" }
+];
+
+type SetURLSearchParams = (
+  nextInit?:
+    | URLSearchParamsInit
+    | ((prev: URLSearchParams) => URLSearchParamsInit),
+  navigateOpts?: NavigateOptions
+) => void;
+
+export const TopologyPopOver = ({
+  size,
+  setSize,
+  topology,
+  searchParams,
+  topologyLabels,
+  setSearchParams
+}: {
+  size: string;
+  topology: any[];
+  topologyLabels: any[];
+  setSize: (v: string) => void;
+  searchParams: URLSearchParams;
+  setSearchParams: SetURLSearchParams;
+}) => {
+  const currentIconRef = useRef();
+
+  const [currentIcon, setCurrentIcon] = useState("");
+
+  const [sortTypes, setSortTypes] = useState<typeof defaultSortTypes>([]);
+
+  useEffect(() => {
+    setSortValues();
+  }, [searchParams, topologyLabels]);
+
+  useEffect(() => {
+    const listener = (event: MouseEvent) => {
+      if (currentIconRef.current?.contains(event.target)) {
```

Comment:
not quire sure I understand what the purpose of currentIcon is for ?

Reply:
The plan is to have both `TopologyPreference` and `TopologySort` to be stateless. So, props are drilled down to handle any kind of events ie. Menu icon and Events inside Popover

Reply:
Why is there any sharing required? The preference and sort popovers should maintain their own state in their own popover.

Reply:
Primarily to have a single handler when clicking outside the active popover. I'll create separate refs and send a commit